OneCommunityGlobal/HGNRest
# Trim White space, sanitize url

# Description
<img width="731" alt="Screen Shot 2023-11-08 at 4 31 54 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/114443664/9d3b906a-200b-4a97-a05b-09cd71a3b35e">


## Related PRS (if any):
N/A

## Main changes explained:
Add third party library to sanitize and trim urls received by userprofileController

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. use postman to test the api http://localhost:4500/api/userProfile/ {your user id here}
4. check that white space in user lastname and firstname are ignored
5. check that white space in url are also ignored
6. check that empty url name will receive an error



## Screenshots or videos of changes:
Before:

https://github.com/OneCommunityGlobal/HGNRest/assets/114443664/a8a1b7eb-75d8-41ee-b893-5f990bf6b98c

After:

https://github.com/OneCommunityGlobal/HGNRest/assets/114443664/03a4a99c-0564-44a4-b4d6-fce5c2532dcd

https://github.com/OneCommunityGlobal/HGNRest/assets/114443664/b9cf345b-f8c9-4c3c-a196-17b5a947ca5a

## Notes
can get the payload of the http request from chrome developer tools by opening the network tab, send an request, and in the payload section, click view source. Then copy it and paste it into postman body->raw
